docs/github/v2/01-getting-started.md

```diff
@@ -137,8 +139,9 @@ jobs:
       - uses: actions/cache@v2
         with:
           path: ${{ matrix.projectPath }}/Library
-          key: Library-${{ matrix.projectPath }}-${{ matrix.targetPlatform }}
+          key: Library-${{ matrix.projectPath }}-${{ matrix.targetPlatform }}-${{ hashFiles(matrix.projectPath) }}
```

Comment:
Is it possible (and better?) to have something like this?

```yaml
hashFiles(matrix.projectPath + 'Assets/**', matrix.projectPath + 'Packages/**', matrix.projectPath + 'ProjectSettings/**')
```

Syntax is probably wrong and it might also not be possible. It's pretty much an open question, but otherwise, this looks good to me 🙌

Reply:
It really depends if you have some stuff in the project directory that doesn't contribute to the creation of `Library`. The simple example might actually be over-pedantic, if anything.

I haven't tried either option, but based on [some googling](https://github.community/t/nested-variable-substitution/140291), at least the following should work:
```
hashFiles(format('{0}/Assets/**', matrix.projectPath), format('{0}/Packages/**', matrix.projectPath), format('{0}/ProjectSettings/**', matrix.projectPath))
```

But the whole thing becomes
```
key: Library-${{ matrix.projectPath }}-${{ matrix.targetPlatform }}-${{ hashFiles(format('{0}/Assets/**', matrix.projectPath), format('{0}/Packages/**', matrix.projectPath), format('{0}/ProjectSettings/**', matrix.projectPath) }}
```

I'm a bit on the fence here regarding whether or not it really makes a big difference. I was thinking that project-at-root-of-repo was more likely to have something extra also at the root of the repo. But I also get that the discrepancy between the two examples is annoying 😆

Reply:
Our filosophy for unity actions is to make setting up CI as easy and fast as possible.

Cache action is always ran immediately after checkout.

It doesn't say explicitly in actions/cache documentation but I think the key is determined only during restore time and reused in the save step (instead of revalidated based on changed files). 

If this is indeed the case then I would argue that your current implementation is rather perfect.